### PR TITLE
Feat/integration tests on actions

### PR DESCRIPTION
**Pull request by DhananjayPurohit**

### CODE REVIEW

File: clients/libs/nodejs/transaction.js [hunk 1]
```diff
@@ -63,8 +63,22 @@ const signFirst = async (clientConfig, signFirstRequestPayload) => {
         socksAgent = { httpAgent: new SocksProxyAgent(torProxy) };
     }
     
-    let response = await axios.post(url, signFirstRequestPayload, socksAgent);
+    let response;
+    try {
+        response = await axios.post(url, signFirstRequestPayload, socksAgent);
+        console.log('Response:', response.data);
```
Comment (ssantos21) [inline clients/libs/nodejs/transaction.js:69]: This can be removed.

```suggestion
```

File: clients/libs/nodejs/transaction.js [hunk 2]
```diff
@@ -88,8 +102,22 @@ const signSecond = async (clientConfig, partialSigRequest) => {
         socksAgent = { httpAgent: new SocksProxyAgent(torProxy) };
     }
 
-    let response = await axios.post(url, partialSigRequest, socksAgent);
+    let response;
+    try {
+        response = await axios.post(url, partialSigRequest, socksAgent);
+        console.log('Response:', response.data);
```
Comment (ssantos21) [inline clients/libs/nodejs/transaction.js:108]: This can be removed.

```suggestion
```

File: clients/libs/nodejs/utils.js [hunk 3]
```diff
@@ -7,17 +7,22 @@ const infoConfig = async (clientConfig, ecl) => {
...
-    let fee_rate_btc_per_kb = await ecl.request('blockchain.estimatefee', [3]); // request(promise)
+    let fee_rate_btc_per_kb;
+    try { 
+        fee_rate_btc_per_kb = await ecl.request('blockchain.estimatefee', [3]); // request(promise)
+    } catch (error) {
+        throw new Error("Error getting fee rate from electrs server");
+    }
 
-    // console.log("fee_rate_btc_per_kb:", fee_rate_btc_per_kb);
+    console.log("fee_rate_btc_per_kb:", fee_rate_btc_per_kb);
```
Comment (ssantos21) [inline clients/libs/nodejs/utils.js:17]: This can be removed.

```suggestion
```

File: clients/libs/nodejs/utils.js [hunk 4]
```diff
@@ -7,17 +7,22 @@ const infoConfig = async (clientConfig, ecl) => {
...
 
     // Why does it happen?
     if (fee_rate_btc_per_kb <= 0) {
         fee_rate_btc_per_kb = 0.00001;
     }
     const fee_rate_sats_per_byte = (fee_rate_btc_per_kb * 100000.0);
 
-    // console.log("fee_rate_sats_per_byte: " + fee_rate_sats_per_byte);
+    console.log("fee_rate_sats_per_byte: " + fee_rate_sats_per_byte);
```
Comment (ssantos21) [inline clients/libs/nodejs/utils.js:25]: This can be removed.

```suggestion
```

File: server/src/endpoints/deposit.rs [hunk 5]
```diff
@@ -183,7 +183,8 @@ pub async fn post_deposit(statechain_entity: &State<StateChainEntity>, deposit_m
         Err(err) => {
             let response_body = json!({
                 "error": "Internal Server Error",
-                "message": err.to_string()
+                "message": err.to_string(),
+                "payload": statechain_id.clone(),
```
Comment (ssantos21) [inline server/src/endpoints/deposit.rs:187]: Why is this ` payload` field being added ?

```suggestion
```

**Review by ssantos21 — CHANGES_REQUESTED** [review]
Sounds good to me, but there are some issues to be addressed.

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (DhananjayPurohit) [reply entrypoint.sh:1]: I wrote that before to create db manually for electrs but not needed anymore. Removing that.

File: server/src/endpoints/deposit.rs (hunk 5 above)
Comment (DhananjayPurohit) [reply server/src/endpoints/deposit.rs:187]: Added that time for debugging, no more needed.

File: clients/libs/nodejs/index.js [hunk 7]
```diff
@@ -15,6 +15,7 @@ const sqlite_manager = require('./sqlite_manager');
 const { v4: uuidv4 } = require('uuid');
 
 const wallet_manager = require('./wallet');
+const path = require('path');
```
[the remark this answers is not in the document]
Comment (DhananjayPurohit) [reply clients/libs/nodejs/index.js:18]: Not needed anymore, Thanks.

File: clients/libs/nodejs/index.js [hunk 8]
```diff
@@ -196,5 +197,6 @@ module.exports = {
     withdrawCoin, 
     newTransferAddress, 
     transferSend,
-    transferReceive
+    transferReceive,
+    getElectrumClient
```
[the remark this answers is not in the document]
Comment (DhananjayPurohit) [reply clients/libs/nodejs/index.js:201]: Not needed anymore, Thanks.

**Pull request merged**